### DISCUSSION TITLE
To add content security policy using addheaders in values.yaml file

### DISCUSSION THREAD

**Discussion by praveensounder** [nginx/kubernetes-ingress#2826] [General]
Currently we are using Nginx ingress controller 2.1.1 - https://github.com/nginxinc/kubernetes-ingress/tree/v2.1.1

We need to add Content security policy to the config file.

We passed using addheaders but its not getting reflected.

Kindly please let us know How to add content security policy in helm charts values.yaml file

We need to add configuration Inside nginx server {} block like below:

add_header Content-Security-Policy "default-src 'self';";

Link: https://content-security-policy.com/examples/nginx/

**Comment by brianehlert** [comment]
This is duplicated here: https://github.com/nginxinc/kubernetes-ingress/issues/2827

**Comment by brianehlert** [comment]
> add_header Content-Security-Policy "default-src 'self';";

Headers are manipulated at the server block level using the VirtualServer resource
https://docs.nginx.com/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#header

**Reply by praveensounder** [reply]
Can we add content security policy in the configmap
